File: frontend/src/services/telegram.ts
import apiClient from './apiClient'

export interface TelegramLinkStatus {
  linked: boolean
  bot_username: string | null
  telegram_username: string | null
  linked_at: string | null
}

export interface TelegramLinkCode {
  code: string
  expires_at: string
  bot_username: string | null
}

export const telegramService = {
  async getStatus(): Promise<TelegramLinkStatus> {
    return apiClient.get('/me/telegram')
  },

  async createLinkCode(): Promise<TelegramLinkCode> {
    return apiClient.post('/me/telegram/link-code')
  },

  async unlink(): Promise<{ message: string }> {
    return apiClient.delete('/me/telegram')
  },
}

export default telegramService

File: backend/src/schemas/telegram.py
"""Schemas for Telegram staff linking."""

from datetime import datetime

from pydantic import BaseModel


class TelegramLinkStatusResponse(BaseModel):
    """Current staff user's Telegram link status."""

    linked: bool
    bot_username: str | None = None
    telegram_username: str | None = None
    linked_at: datetime | None = None


class TelegramLinkCodeResponse(BaseModel):
    """One-time Telegram link code."""

    code: str
    expires_at: datetime
    bot_username: str | None = None
